Unable to use ximgproc's structured edge detection on iOS

I'm unable to use the structured edge detection on iOS as there seems to be a bug when working with compressed files. The model provided with the ximgproc module is a .yml.gz file. A quick inspection of `persistence.cpp` showed that `USE_ZLIB`gets undefined on iOS at the very beginning of the file leading to the following error. The cmake logs show that zlib builds and gets linked into the iOS framework fine, though.

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => current master
- Operating System / Platform => macOS 12.2.2 (host), iOS 10.2 (target)
- Compiler => clang (Xcode 8.2.1 default)

##### Detailed description

I'm getting the following error log in Xcode:
```
OpenCV Error: The function/feature is not implemented (There is no compressed file storage support in this configuration) in cvOpenFileStorage, file /Users/xyz/opencv-master/modules/core/src/persistence.cpp, line 4235
2017-01-31 11:17:47.500052 mytestproject[1226:473262] *** Terminating app due to uncaught exception '/Users/xyz/opencv-master/modules/core/src/persistence.cpp:4235: error: (-213) There is no compressed file storage support in this configuration in function cvOpenFileStorage
```

##### Steps to reproduce
The model file taken from the ximgproc module can be downloaded [here](https://www.dropbox.com/s/q78tbnkqatae16s/model.yml.gz?dl=0)

```cpp
            Mat img = imread("image.png");
            Mat imgFloat;
            img.convertTo(imgFloat, cv::DataType<float>::type, 1 / 255.0);
            Mat rfEdges(imgFloat.size(), imgFloat.type());

            String modelFile = "./model.yml.gz";
            cv::Ptr<cv::ximgproc::StructuredEdgeDetection> edgeDetector = cv::ximgproc::createStructuredEdgeDetection(modelFile);
            edgeDetector->detectEdges(imgFloat, rfEdges);
```

--- a/modules/core/src/persistence.cpp
+++ b/modules/core/src/persistence.cpp
@@ -50,15 +50,6 @@
 
 #define USE_ZLIB 1
 
-#ifdef __APPLE__
-#  include "TargetConditionals.h"
-#  if (defined TARGET_OS_IPHONE && TARGET_OS_IPHONE) || (defined TARGET_IPHONE_SIMULATOR && TARGET_IPHONE_SIMULATOR)
-#    undef USE_ZLIB
-#    define USE_ZLIB 0
-     typedef void* gzFile;
-#  endif
-#endif
-
 #if USE_ZLIB
 #  ifndef _LFS64_LARGEFILE
 #    define _LFS64_LARGEFILE 0

